File: app/src/main/java/com/dk/tengerms/views/fragments/ChangePassword.java
package com.dk.tengerms.views.fragments;

import android.os.Bundle;
import android.os.Handler;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.dk.tengerms.R;
import com.dk.tengerms.controllers.launcher;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.fragment.NavHostFragment;


public class ChangePassword extends Fragment implements View.OnClickListener {
    Button bt_login;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_change_pwd, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        init(view);
    }

    public void init(View view) {
        bt_login = view.findViewById(R.id.bt_login);
        bt_login.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.bt_login:
                login();
                break;

        }
    }

    private void login() {
        ((launcher)getActivity()).globalClass.showDialog(getActivity());
        final Handler handler = new Handler();
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {
                ((launcher)getActivity()).globalClass.hideLoader();
                NavHostFragment.findNavController(ChangePassword.this).navigate(R.id.login_to_dashboard);
            }
        }, 1500);
    }
}

File: app/src/main/java/com/paxees/tcc/views/fragments/settingsFragments/ChangePassword.kt
package com.paxees.tcc.views.fragments.settingsFragments

import android.os.Bundle
import android.text.TextUtils
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.fragment.findNavController
import com.paxees.tcc.R
import com.paxees.tcc.controllers.CIFRootActivity
import com.paxees.tcc.network.ResponseHandlers.callbacks.ChangePasswordCallBack
import com.paxees.tcc.network.enums.RetrofitEnums
import com.paxees.tcc.network.networkmodels.request.ChangePasswordRequest
import com.paxees.tcc.network.networkmodels.response.baseResponses.BaseResponse
import com.paxees.tcc.network.networkmodels.response.baseResponses.ChangePasswordResponse
import com.paxees.tcc.network.store.TCCStore
import com.paxees.tcc.utils.SessionManager
import com.paxees.tcc.utils.ToastUtils
import kotlinx.android.synthetic.main.fragment_changedpwd.*
import kotlinx.android.synthetic.main.fragment_login.*
import kotlinx.android.synthetic.main.toolbar_theme.*

class ChangePassword : Fragment(), View.OnClickListener {
    var sessionManager: SessionManager? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_changedpwd, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        init(view)
    }

    fun init(view: View?) {
        sessionManager = SessionManager(activity)
        backBtn.setOnClickListener(this)
        bt_submit.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.backBtn -> {
                findNavController().popBackStack()
            }
            R.id.bt_submit -> {
                if (validation()) {
                    changePassword()
                }
            }
        }
    }

    fun validation(): Boolean {
        val pwd = etNewPassword!!.text.toString().trim { it <= ' ' }
        val confirmPwd = etConfirmPassword!!.text.toString().trim { it <= ' ' }
        return if (TextUtils.isEmpty(pwd)) {
            etNewPassword!!.error = "New Password should not be empty"
            etNewPassword!!.requestFocus()
            false
        } else if (TextUtils.isEmpty(confirmPwd)) {
            etConfirmPassword!!.error = "Confirm New Password should not be empty"
            etConfirmPassword!!.requestFocus()
            false
        } else if (pwd != confirmPwd) {
            etNewPassword!!.error = "New & Confirm New Password should be same"
            etNewPassword!!.requestFocus()
            false
        } else {
            true
        }
    }


    fun changePassword() {
        var userid = (activity as CIFRootActivity).sharedPreferenceManager.customerDetails[0].id
        var request = ChangePasswordRequest()
        request.password = etConfirmPassword.text.toString().trim()
        (activity as CIFRootActivity?)!!.globalClass!!.showDialog(activity)
        TCCStore.instance!!.changePassword(
            RetrofitEnums.URL_HBL,
            userid.toString(),
            request,
            object :
                ChangePasswordCallBack {
                override fun Success(response: ChangePasswordResponse) {
                    ToastUtils.showToastWith(activity, "Password has been updated")
                    findNavController().popBackStack()
                    (activity as CIFRootActivity?)!!.globalClass!!.hideLoader()
                }

                override fun Failure(baseResponse: BaseResponse) {
                    ToastUtils.showToastWith(activity, baseResponse.message, "")
                    (activity as CIFRootActivity?)!!.globalClass!!.hideLoader()
                }
            })
    }
}